docs/validate.ts

```diff
@@ -0,0 +1,92 @@
+import { MdxJsxAttribute, MdxJsxFlowElement, MdxJsxTextElement } from 'mdast-util-mdx';
+
+// A type guard to check if a node has children
+function isParent(node: Node | Parent): node is Parent {
+  return 'children' in node;
+}
+
+// Pure function to return an array of all nodes in the tree
+function flattenTree(node: Node | Parent): (Node | Parent)[] {
+  const nodes = [node];
+  if (isParent(node)) {
+    return nodes.concat(node.children.flatMap(flattenTree));
+  }
+  return nodes;
+}
+
+// Pure predicate function to check if node is a TutorialFallback element
+function isTutorialFallbackNode(node: Node): node is MdxJsxFlowElement | MdxJsxTextElement {
+  const isFlowElement = node.type === 'mdxJsxFlowElement';
+  const isJsxTextElement = node.type === 'mdxJsxTextElement';
+  const isNameTutorialFallback = 'name' in node && node.name === 'TutorialFallback';
+  
+  return (isFlowElement || isJsxTextElement) && isNameTutorialFallback;
+}
+
+// Pure function to extract filename from node attributes
+function extractFilename(node: MdxJsxFlowElement | MdxJsxTextElement): string | null {
+  const fileNameAttr = node.attributes?.find(
+    attr => attr.type === 'mdxJsxAttribute' && attr.name === 'filename'
+  ) as MdxJsxAttribute | undefined;
+  
+  return (fileNameAttr?.value as string) || null;
+}
+
+export function extractArgumentFileNames(mdxContent: Value): (string | null)[] {
+  const tree = fromMarkdown(mdxContent, {
+    extensions: [mdx()],
+    mdastExtensions: [mdxFromMarkdown()]
+  });
+
+  return flattenTree(tree)
+    .filter(isTutorialFallbackNode)
+    .map(extractFilename);
+}
+
+function validateResourceAccess(mdxContent: Value, validFilenames: string[]): boolean {
+  const argFilenames = extractArgumentFileNames(mdxContent);
+  const validResources = argFilenames.map(argFileName =>
+    validFilenames.includes(argFileName as string)
+  );
+
+  validResources.forEach((isValid, index) => {
+    if (!isValid) {
+      console.log(`\tERROR: NON-EXISTENT STATIC RESOURCE:\t "${argFilenames[index]}"`);
+    }
+  });
+
+  return validResources.every(isValid => isValid);
+}
+
+
+module.exports = {
+  extractArgumentFileNames,
+  validateResourceAccess,
+};
+
+if (require.main === module) {
+  const glob = require('glob');
+  const path = require('path');
+  const fs = require('fs');
+
+  const staticResources = glob.sync('static/**/*.json');
+  const validResourceFilenames = staticResources.map((filePath: string) =>
+      path.basename(filePath, path.extname(filePath))
+  );
+  const mdxFilePaths = glob.sync('docs/**/*.{md,mdx}');
+  let validAccesses = mdxFilePaths
+    .map((testFilePath: string) => fs.readFileSync(testFilePath, 'utf8'))
+    .map((mdxContent: Value) => validateResourceAccess(mdxContent, validResourceFilenames));
+
+  validAccesses.forEach((validAccess: boolean, index: number) => { 
+    if (!validAccess) {
+      throw new Error(`Invalid static resource access in ${mdxFilePaths[index]}`)
+    }
+  });
```

Comment:
If I'm understanding correctly (please correct me if I'm wrong), when you mentioned better error logging, you were saying that the previous code shows all invalid accesses, unlike this newer code, which shows only the first invalid access.

This can be solved by using `map` to combine all invalid accesses into a single message before logging.


For example:
```ts
const erroneousFilePaths = validAccesses
    .flatMap((validAccess, index) => validAccess ? [] : [mdxFilePaths[index]])

if(errorneousFilePaths.length > 0) {
    throw new Error(errorneousFilePaths.join("\n"))
}
```

Reply:
Wait, I think the code can be further improved.